docs/proposal/dns_resolve.md

```diff
@@ -0,0 +1,218 @@
+---
+title: Your short, descriptive title
+authors:
+- "@zhxuzhonghu"
+reviewers:
+- 
+approvers:
+- 
+
+
+creation-date: 2024-05-08
+
+---
+
+## Support DNS Resolution in Cluster Manager
+
+<!--
+This is the title of your KEP. Keep it short, simple, and descriptive. A good
+title can help communicate what the KEP is and should be considered as part of
+any review.
+-->
+
+### Summary
+
+<!--
+This section is incredibly important for producing high-quality, user-focused
+documentation such as release notes or a development roadmap.
+A good summary is probably at least a paragraph in length.
+-->
+
+Envoy supports many different cluster types, including `Strict DNS`, `Logical DNS`. However, given to Kmesh works in the kernel with ebpf. Previously Kmesh does not either of the DNS typed clusters. For traffic matches these kind of cluster, it will be dropped. 
+
+In this propsosal, I would suggest to improve Kmesh to support DNS types cluster, so we can support all kinds of clusters afterwards.
+
+
+### Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users.
+-->
+
+In istio, [External Name service](https://kubernetes.io/docs/concepts/services-networking/service/#externalname) and DNS resolution typed [ServiceEntry](https://istio.io/latest/docs/reference/config/networking/service-entry/#ServiceEntry-Resolution) are widely used. For both kind of configs, istiod will generate associated DNS typed clusters.
+
+So many people have depend on this kind services, Kmesh have to support it to make people migrate to it seamlessly.
+
+Suppose we create a ServiceEntry like below:
+
+```yaml
+apiVersion: networking.istio.io/v1
+kind: ServiceEntry
+metadata:
+  name: se
+  namespace: default
+spec:
+  hosts:
+  - news.google.com
+  ports:
+  - name: port1
+    number: 80
+    protocol: HTTP
+  resolution: DNS
+```
+
+It will result into a cluster below:
+
+```json
+{
+    "name": "outbound|80||news.google.com",
+    "type": "STRICT_DNS",
+    "connectTimeout": "10s",
+    "lbPolicy": "LEAST_REQUEST",
+    "loadAssignment": {
+        "clusterName": "outbound|80||news.google.com",
+        "endpoints": [
+            {
+                "locality": {},
+                "lbEndpoints": [
+                    {
+                        "endpoint": {
+                            "address": {
+                                "socketAddress": {
+                                    "address": "news.google.com",
+                                    "portValue": 80
+                                }
+                            }
+                        },
+                        "metadata": {
+                            "filterMetadata": {
+                                "istio": {
+                                    "workload": ";;;;"
```

Comment:
What is the meaning of this line?

Reply:
This is a composed string including some metadata from pod. If these are empty, then it will be this type